### ISSUE TITLE
The HP LaserJet M111a

### ISSUE THREAD

**Issue by debiantriage** [OpenPrinting/cups-filters#514]
I have [recently been involved ](https://forums.linuxmint.com/viewtopic.php?t=392149) in some use interaction. A long and twisty series of posts.

The user has an HP LaserJet M111a. The device is USB-only but supports IPP-over-USB. The problem is that the everywhere model does not produce a working print queue.

We have the printer attributes. My contention is that they do not contain enough for using Apple or PWG raster. Am I correct?

Files attached. Sorry, I do not think this is a cups-filters issue but did not know where to put it.

[M111a.lsub.txt](https://github.com/OpenPrinting/cups-filters/files/10902653/M111a.lsub.txt)
[M111a.avahi-browse.txt](https://github.com/OpenPrinting/cups-filters/files/10902656/M111a.avahi-browse.txt)
[M111a.attrs.txt](https://github.com/OpenPrinting/cups-filters/files/10902659/M111a.attrs.txt)

**Comment by tillkamppeter** [comment]
Principally, the printer supports PCLm and PWG Raster. PCLm is not supported by CUPS, but by cups-filters, via the `driverless` utility.

The problem is that the response to the `get-printer-attributes` IPP request does not contain a `document-format-supported` attribute, so CUPS or the `driverless` utility do not discover the printer's PWG Raster or PCLm support.

The DNS-SD record of the printer reports both PWG Raster and PCLm support properly.

@michaelrsweet is this standards-conforming? Do we need to correct CUPS and/or cups-filters to make this printer working?

**Comment by wifiprintguy** [comment]
The firmware appears to be out of date - the [M111a.attrs.txt](https://github.com/OpenPrinting/cups-filters/files/10902659/M111a.attrs.txt) lists the following:

printer-firmware-name (nameWithoutLanguage) = 20210805
printer-firmware-string-version (textWithoutLanguage) = 20210805
printer-firmware-version (octetString) = 20210805

The latest firmware on the HP site is 20221107:

https://support.hp.com/us-en/drivers/selfservice/hp-laserjet-m109a-m112a-printer-series/30190513/model/34443746

Not sure if updating the firmware will resolve the lack of "document-format-supported", which ought to always be supported by the printer and supplied for the "all" group.

**Comment by debiantriage** [comment]
Thank you @tillkamppeter  @wifiprintguy 

I will draw the user's attention to both your replies but will advise that he install HPLIP for his printing needs.

My interest in this device was partially piqued by [HP's Customer Support](https://support.hp.com/us-en/document/c07937244) for it. 

> Mobile printing services USB
   Print languages PCLm/PCLmS, URF, PWG

I had not seen either of these statements before for a USB-only printer. It seemed to imply a support for IPP-over-USB. Note that the promise of Apple raster is not indicated in the TXT record.

**Comment by michaelrsweet** [comment]
@tillkamppeter No, omitting the "document-format-supported" attribute is not standards-conforming. As for the TXT record, that is generated by the IPP-USB daemon on the system (whichever version that is) and not by the printer...

From the [About Airprint](https://support.apple.com/en-ca/HT201311) page, it looks like only the M111w (wireless) model is AirPrint certified, not the USB-only (M111a) model.

**Comment by EHT-droid** [comment]
> Principally, the printer supports PCLm and PWG Raster. PCLm is not supported by CUPS, but by cups-filters, via the `driverless` utility.
> 
> The problem is that the response to the `get-printer-attributes` IPP request does not contain a `document-format-supported` attribute, so CUPS or the `driverless` utility do not discover the printer's PWG Raster or PCLm support.
> 
> The DNS-SD record of the printer reports both PWG Raster and PCLm support properly.
> 
> @michaelrsweet is this standards-conforming? Do we need to correct CUPS and/or cups-filters to make this printer working?

I am the person whose HP LasetJetM111a isn't working. Our test print is stopped because of a configuration error (". . . the scheduler could not execute a filter"). In the printer's "Job attributes" dropdown, under "job.printer.state.reasons" it gives the value: ["cups.missing.filter.warning"]. Could this mean that, as you've suggested, the cups-filters need to be adjusted to get this printer working?
